### ISSUE TITLE
alt-left/right overwrites fish_prompt without clearing trailing characters

### ISSUE THREAD

**Issue by artagnon** [fish-shell/fish-shell#3550]
I have a simple prompt:
```
function fish_prompt
	set_color yellow
	echo (pwd) | sed -e "s|^$HOME|~|" -e "s|^/sandbox/rramacha|\$s|"
	set_color normal
	echo '$ '
end
```

When I alt-left from `~/src/llvm/tools/polly/test/ScopDetect` to `~/src/jsoncpp`, my prompt is mangled:
```
~/src/jsoncppols/polly/test/ScopDetect
$
```

Not sure how to work around this except by padding the path with trailing spaces?

**Comment by faho** [comment]
This seems related to #3537, since it also only seems to occur with a multi-line prompt.

**Comment by artagnon** [comment]
I looked at the that thread earlier, but it didn't strike me as being related. This is probably a dupe, and the most elegant workaround is a single-line prompt.

**Comment by artagnon** [comment]
Can I change the color of the prompt twice on the same line? I want to distinguish between the cwd path and '$'.

**Comment by krader1961** [comment]
Yes, you can change colors multiple times. I do it:

<img width="373" alt="screen shot 2016-11-14 at 12 06 46" src="https://cloud.githubusercontent.com/assets/7545917/20280823/eaf7d660-aa62-11e6-8d74-f3377c3ac233.png">

**Comment by artagnon** [comment]
Hm, I added `-n` to my `echo` invocations, but it seems to have no effect.

**Comment by krader1961** [comment]
Works for me:

```
function fish_prompt
    set -l cwd (echo (pwd) | sed -e "s|^$HOME|~|" -e "s|^/sandbox/rramacha|\$s|")
    echo -n (set_color yellow)$cwd(set_color normal) '$ '
end
```

Note that mutating the PWD can be done more efficiently but I'll leave that as an exercise for the student. Hint: start by using `$PWD` rather than `(pwd)` and use the `string` builtin rather than shelling out to `sed`.

**Comment by artagnon** [comment]
Thanks! Much appreciated.

**Comment by faho** [comment]
Can't reproduce in master, there've been a *lot* of changes since this was reported. Closing!

**Comment by zanchey** [comment]
It's broken in 3.1.2 but fixed in 3.2.0.
